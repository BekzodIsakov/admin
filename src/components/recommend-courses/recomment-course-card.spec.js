import { screen } from '@testing-library/react';
import { RecommendCourses } from './index';
import { customRender } from '@test-utils/helpers';

const items = [
  {
    title: {
      uz: 'Test'
    },
    picture: {
      url: ''
    },
    description: {
      uz: ''
    },
    slug: ''
  }
];

describe('recommend-courses-component', () => {
  test('render empty component', () => {
    const { asFragment } = customRender(
      <RecommendCourses
        recommendedCourses={undefined}
        data-testid='recommend-course'
      />
    );

    const ancestor = screen.queryByTestId('recommend-course');
    const descendant = screen.getByRole('heading', {
      level: 3,
      name: /Hozircha hech qanday tavsiya qilinadigan kurslar yoq/i
    });

    expect(asFragment()).toMatchSnapshot();
    expect(ancestor).toContainElement(descendant);
  });

  test('render component with items', () => {
    const { asFragment } = customRender(
      <RecommendCourses
        recommendedCourses={items}
        data-testid='recommend-course'
      />
    );

    const ancestor = screen.queryByTestId('recommend-course');
    const descendant = screen.getByText(items[0].title.uz);

    expect(asFragment()).toMatchSnapshot();
    expect(ancestor).not.toHaveTextContent(
      /Hozircha hech qanday tavsiya qilinadigan kurslar yoq/i
    );
    expect(ancestor).toContainElement(descendant);
  });
});
