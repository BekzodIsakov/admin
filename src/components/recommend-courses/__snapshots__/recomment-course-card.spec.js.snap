// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recommend-courses-component render component with items 1`] = `
<DocumentFragment>
  .c2::-webkit-scrollbar {
  width: 0;
}

.c0 {
  padding: 20px 24px 52px 24px;
  background-color: #fff;
  border-radius: 4px;
  margin-top: 20px;
}

.c1 {
  font-style: normal;
  font-weight: bold;
  font-size: 20px;
  line-height: 24px;
  -webkit-letter-spacing: -0.02em;
  -moz-letter-spacing: -0.02em;
  -ms-letter-spacing: -0.02em;
  letter-spacing: -0.02em;
  color: #000000;
  margin-bottom: 24px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c10 {
  width: 12px;
  height: 12px;
  fill: #BA8D5B;
  stroke: #BA8D5B;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
}

.c10[disabled] {
  opacity: 0.2;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
  width: 32%;
}

.c6 {
  width: 100%;
  height: 158px;
  border-radius: 4px;
  object-fit: cover;
  margin-bottom: 16px;
  object-position: right center;
}

.c7 {
  font-style: normal;
  font-weight: bold;
  font-size: 18px;
  line-height: 22px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #000000;
  margin-bottom: 6px;
}

.c8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-style: normal;
  font-weight: normal;
  font-size: 1.4rem;
  line-height: 16px;
  -webkit-letter-spacing: -0.02em;
  -moz-letter-spacing: -0.02em;
  -ms-letter-spacing: -0.02em;
  letter-spacing: -0.02em;
  color: rgba(0,0,0,0.5);
}

.c9 {
  font-style: normal;
  font-weight: bold;
  font-size: 14px;
  line-height: 18px;
  -webkit-letter-spacing: -0.02em;
  -moz-letter-spacing: -0.02em;
  -ms-letter-spacing: -0.02em;
  letter-spacing: -0.02em;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 19px 0;
  color: #ba8d5b;
}

.c9 svg {
  margin-left: 12px;
  -webkit-transition: all ease 0.3s;
  transition: all ease 0.3s;
}

.c9:hover svg {
  margin-left: 18px;
}

@media only screen and (max-width:1024px) {
  .c2 {
    overflow: auto;
    max-width: 800px;
    padding-bottom: 0 !important;
  }
}

@media only screen and (max-width:1024px) {
  .c0 {
    padding-right: 0;
    padding-bottom: 0;
  }
}

@media only screen and (max-width:1024px) {
  .c3 {
    width: 800px;
    overflow-x: scroll;
    padding-bottom: 0 !important;
  }

  .c3::-webkit-scrollbar {
    width: 0;
  }
}

@media only screen and (max-width:1024px) {
  .c4 {
    padding-right: 20px;
  }
}

<div
    class="c0"
    data-testid="recommend-course"
  >
    <h3
      class="c1"
    >
      Sizga tavsiya qilamiz
    </h3>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <img
              class="c6"
              data-testid="recommended-course-img"
              src=""
            />
            <h4
              class="c7"
              data-test-id="recommended-course-title"
            >
              Test
            </h4>
            <p
              class="c8"
              data-testid="recommended-course-text"
            >
              ...
            </p>
            <a
              class="c9"
              data-testid="recommended-courseLink"
              href="https://landing.najottalim.uz/courses/"
              target="_blank"
            >
              Batafsil
              <svg
                class="c10"
              >
                <use
                  xlink:href="sprite.svg#arrow_right-icon"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`recommend-courses-component render empty component 1`] = `
<DocumentFragment>
  .c2::-webkit-scrollbar {
  width: 0;
}

.c0 {
  padding: 20px 24px 52px 24px;
  background-color: #fff;
  border-radius: 4px;
  margin-top: 20px;
}

.c1 {
  font-style: normal;
  font-weight: bold;
  font-size: 20px;
  line-height: 24px;
  -webkit-letter-spacing: -0.02em;
  -moz-letter-spacing: -0.02em;
  -ms-letter-spacing: -0.02em;
  letter-spacing: -0.02em;
  color: #000000;
  margin-bottom: 24px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media only screen and (max-width:1024px) {
  .c2 {
    overflow: auto;
    max-width: 800px;
    padding-bottom: 0 !important;
  }
}

@media only screen and (max-width:1024px) {
  .c0 {
    padding-right: 0;
    padding-bottom: 0;
  }
}

@media only screen and (max-width:1024px) {
  .c3 {
    width: 800px;
    overflow-x: scroll;
    padding-bottom: 0 !important;
  }

  .c3::-webkit-scrollbar {
    width: 0;
  }
}

@media only screen and (max-width:1024px) {
  .c4 {
    padding-right: 20px;
  }
}

<div
    class="c0"
    data-testid="recommend-course"
  >
    <h3
      class="c1"
    >
      Sizga tavsiya qilamiz
    </h3>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <h3>
            Hozircha hech qanday tavsiya qilinadigan kurslar yoq
          </h3>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
